src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -8,10 +8,12 @@
 #include <util/string_utils.h>
 
 smt2_incremental_decision_proceduret::smt2_incremental_decision_proceduret(
-  std::string _solver_command)
-  : solver_command{std::move(_solver_command)},
+  std::string _solver_command,
+  message_handlert &message_handler)
+  : solver_command(std::move(_solver_command)),
     number_of_solver_calls{0},
-    solver_process{split_string(solver_command, ' ', false, true)}
+    solver_process{split_string(solver_command, ' ', false, true)},
```

Comment:
The split_string looks brittle.

Reply:
The arguments are provided to `cbmc` in space separated form. Therefore they currently need to be split in order separate them. Any suggestions for making this less brittle?